diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7d0eb9b336..5d5fbed798 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -44,7 +44,7 @@ if (${srcdir} STREQUAL ${bindir})
 endif (${srcdir} STREQUAL ${bindir})
 
 # Define minimum CMake version required
-cmake_minimum_required (VERSION 2.8.5)
+cmake_minimum_required (VERSION 2.8.5...3.0)
 
 # Use NEW behavior with newer CMake releases
 foreach(p
