Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 20368)
+++ CMakeLists.txt	(working copy)
@@ -49,6 +49,7 @@
 # Use NEW behavior with newer CMake releases
 foreach(p
 		CMP0025 # CMake 3.0: Compiler id for Apple Clang is now AppleClang
+        CMP0074 # CMake 3.12: Use the Packagename_ROOT variables
 		)
 	if(POLICY ${p})
 		cmake_policy(SET ${p} NEW)
