diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4b85bc99ad..aaa87397d0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -230,11 +230,11 @@ else (CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
 	set (ACCELERATE_NEW_LAPACK 0)
 endif (CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
 
-if (HAVE_LAPACK AND ACCELERATE_NEW_LAPACK GREATER 0)
+if (HAVE_LAPACK AND ACCELERATE_NEW_LAPACK GREATER 0 AND LAPACK_LIBRARIES MATCHES ".*Accelerate.*")
 	# Recent macOS with Aarwin kernel > 22.4 needs this compiler flag:
 	add_compile_definitions(ACCELERATE_NEW_LAPACK)
 	message("-- Must add compiler flag -DACCELERATE_NEW_LAPACK definition for macOS kernel version ${CMAKE_HOST_SYSTEM_VERSION}")
-endif (HAVE_LAPACK AND ACCELERATE_NEW_LAPACK GREATER 0)
+endif (HAVE_LAPACK AND ACCELERATE_NEW_LAPACK GREATER 0 AND LAPACK_LIBRARIES MATCHES ".*Accelerate.*")
 
 if (NOT DEFINED GMT_EXCLUDE_BLAS)
 	find_package (BLAS)
