diff --git a/src/gmt_notposix.h b/src/gmt_notposix.h
index 586c88d136..bdc9a006fc 100644
--- a/src/gmt_notposix.h
+++ b/src/gmt_notposix.h
@@ -331,11 +331,9 @@
 
 #ifndef NAN
 #	ifdef _MSC_VER
-#		include <ymath.h>
-#		define NAN _Nan._Double
+#		define NAN (-(float)(((float)(1e+300 * 1e+300)) * 0.0F));
 #	else /* _MSC_VER */
-		static const double _NAN = (HUGE_VAL-HUGE_VAL);
-#		define NAN _NAN
+#		define NAN (HUGE_VAL-HUGE_VAL);
 #	endif /* _MSC_VER */
 #endif /* !NAN */
 
