--- gmt-5.4.2.orig/cmake/dist/CMakeLists.txt	2017-01-16 19:36:06.000000000 -0200
+++ gmt-5.4.2/cmake/dist/CMakeLists.txt	2017-07-18 13:36:22.253463867 -0300
@@ -167,16 +167,6 @@
 		install (PROGRAMS ${OMP_DLL_LOC} DESTINATION ${GMT_BINDIR} COMPONENT Runtime)
 	endif (OMP_DLL_LOC)
 
-	if (WIN32 AND TRUE) 
-		# Install the gmtmex on Windows. Need to make this work via variables in ConfigUser.cmake
-		if (BITAGE EQUAL 64)
-			install (PROGRAMS "C:/progs_cygw/GMTdev/gmt-mex/trunk/src/gmtmex.mexw64" DESTINATION ${GMT_BINDIR} COMPONENT Runtime)
-		else ()
-			install (PROGRAMS "C:/progs_cygw/GMTdev/gmt-mex/trunk/src/gmtmex.mexw32" DESTINATION ${GMT_BINDIR} COMPONENT Runtime)
-		endif ()
-		install (PROGRAMS "C:/progs_cygw/GMTdev/gmt-mex/trunk/src/gmt.m" DESTINATION ${GMT_BINDIR} COMPONENT Runtime)
-	endif (WIN32 AND TRUE)
-
 	# Fix bundle:
 	install (CODE "
 	if (CMAKE_INSTALL_PREFIX MATCHES \"_CPack_Packages\")
